website/src/pages/docs/features/explicit-resource-management.mdx

```diff
@@ -0,0 +1,230 @@
+                }
+            }
+        })
+    });
+    await yoga.dispose();
+    console.log('Yoga is disposed')
+    ```
+    </Tabs.Tab>
+</Tabs>
+
+In the first example, we use the `await using` syntax to create a new instance of `yoga` and dispose
+of it when the block is exited. In the second example, we create a new instance of `yoga` and
+dispose of it using the `dispose` method.
+
+### Dispose on Node.js
+
+When running GraphQL Yoga on Node.js, you can use process event listeners or server's `close` event
+to trigger GraphQL Yoga's disposal. Or you can configure GraphQL Yoga to handle this automatically
+by listening `process` exit signals.
+
+<Tabs items={['Explicit disposal', 'Automatic disposal on process exit']}>
+
+<Tabs.Tab>
+
+We can dispose of the GraphQL Yoga instance when the server is closed like below.
+
+```ts
+import { createServer } from 'http'
+import { createSchema, createYoga } from 'graphql-yoga'
+
+const yoga = createYoga({
+  schema: createSchema({
+    typeDefs: /* GraphQL */ `
+      type Query {
+        hello: String
+      }
+    `,
+    resolvers: {
+      Query: { hello: () => 'world' }
+    }
+  })
+})
+
+const server = createServer(yoga)
+server.listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+})
+server.once('close', async () => {
+  await yoga.dispose()
+  console.info('Server is disposed so is Yoga')
+})
+```
+
+</Tabs.Tab>
+
+<Tabs.Tab>
+
+`disposeOnProcessTerminate` option will register an event listener for `process` termination in
+Node.js
+
+```ts
+import { createServer } from 'http'
+import { createSchema, createYoga } from 'graphql-yoga'
+
+createServer(
+  createYoga({
+    schema: createSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          hello: String
+        }
+      `,
+      resolvers: { Query: { hello: () => 'world' } }
+    }),
+    disposeOnProcessTerminate: true
+  })
+).listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+})
+```
+
+</Tabs.Tab>
+
+</Tabs>
+
+## Dispose Plugins
+
+If you have plugins that need some internal resources to be disposed of, you can use the `onDispose`
+hook to dispose of them. This hook will be invoked when the GraphQL Yoga instance is disposed like
+above.
+
+```ts
+let dbConnection: Connection
+const plugin = {
+  onPluginInit: async () => {
+    dbConnection = await createConnection()
+  },
+  onDispose: async () => {
+    // Dispose of resources
+    await dbConnection.close()
+  }
+}
+```
+
+Or you can flush a queue of operations when the server is shutting down.
+
+```ts
+const backgroundJobs: Promise<void>[] = []
+
+const plugin = {
+  onRequestParse() {
+    backgroundJobs.push(
+      sendAnalytics({
+        /* ... */
+      })
+    )
+  },
+  onDispose: async () => {
+    // Flush the queue of background jobs
+    await Promise.all(backgroundJobs)
+  }
+}
+```
+
+But for this kind of purposes, `waitUntil` can be a better choice.
```

Comment:
I feel like this should mainly be covered within the `envelop-plugins` section 🤔

Reply:
Maybe we can have in both because here we also show the difference between `onDispose` and `waitUntil`.

Reply:
I added that information in the plugins section, too. Let me know what you think.

Reply:
looking good! thank you